lib/src/realm_class.dart

```diff
@@ -483,6 +483,40 @@ class Realm implements Finalizable {
       throw RealmError('Starting a write transaction on a frozen Realm is not allowed.');
     }
   }
+
+  /// Compacts a Realm file. A Realm file usually contains free/unused space.
+  ///
+  /// This method removes this free space and the file size is thereby reduced.
+  /// Objects within the Realm file are untouched.
+  /// Note: The file system should have free space for at least a copy of the Realm file. This method must not be called inside a transaction.
+  /// The Realm file is left untouched if any file operation fails.
+  static Future<bool> compact(Configuration config) {
+    late Configuration compactConfig;
+    if (config is LocalConfiguration) {
```

Comment:
I'm wondering why we need to recreate the configs before compact.

Reply:
the idea is not to invoke any callbacks while opening from within this method. We copy only the relevant fields in the new configs

Reply:
It is because `disableFormatUpgrade` and `isReadOnly` is set differently

Reply:
wonder if we should not throw if `isReadOnly` is set?